lib/kbsemi.gi

```diff
@@ -501,33 +501,28 @@ InstallMethod(IsReduced,
 "for a Knuth Bendix rewriting system", true,
 [IsKnuthBendixRewritingSystem and IsKnuthBendixRewritingSystemRep and IsMutable], 0,
 function(kbrws)
-  local i,copy_of_kbrws,u;
+  local i, copy_of_kbrws, u, tzr;
 
   # if the rws already knows it is reduced return true
   if kbrws!.reduced then return true; fi;
 
-  for i in [1..Length(Rules(kbrws))] do
+  tzr := TzRules(kbrws);
+  for i in [1..Length(tzr)] do
 
-    u := Rules(kbrws)[i];
+    u := tzr[i];
```

Comment:
Could you briefly explain or at least hint what issue this fixes, and why it is correct (e.g. in the commit message)?